synapse/handlers/user_directory.py

```diff
@@ -464,6 +464,10 @@ async def _handle_possible_remote_profile_change(
 
         prev_name = prev_event.content.get("displayname")
         new_name = event.content.get("displayname")
+
+        # Replace any NULL characters in the name as these cannot be stored in the database
+        new_name = new_name.replace("\x00", "\uFFFD")
```

Comment:
We used a space as the replacement character in #10820. I think this might have been chosen to make sure that the text-search stuff (as in https://www.postgresql.org/docs/current/datatype-textsearch.html and https://www.postgresql.org/docs/current/textsearch-controls.html) work more nicely. Let me see if I can experiment to see how postgres handles a replacement character in that context...